Add ContainerProvider for resolving dependencies in XAML

chowarth opened the pull request:
﻿## Description of Change

Resolves https://github.com/PrismLibrary/Prism/issues/3238 by adding `ContainerProvider<T>` based on the previous implementation in `Prism.Forms`.

### Bugs Fixed

N/A

### API Changes

Added:

- `ContainerProvider<T>`


### Behavioral Changes

N/A

### PR Checklist

- [X] Has tests (if omitted, state reason in description)
- [X] Rebased on top of master at time of PR
- [X] Changes adhere to coding standard

dansiegel commented on tests/Maui/Prism.DryIoc.Maui.Tests/Prism.DryIoc.Maui.Tests.csproj:
```diff
@@ -32,4 +32,10 @@
     <ProjectReference Include="..\..\..\src\Maui\Prism.Maui.Rx\Prism.Maui.Rx.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <MauiXaml Update="Mocks\Views\MockXamlView.xaml">
+      <Generator>MSBuild:Compile</Generator>
+    </MauiXaml>
+  </ItemGroup>
+
```
This is the result of a bug in Visual Studio. Please remove this block.

chowarth commented on src/Maui/Prism.Maui/Ioc/ContainerProvider.cs:
```diff
@@ -0,0 +1,65 @@
+    /// <summary>
+    /// The Name used to register the type with the Container
+    /// </summary>
+    public string Name { get; set; }
```
Thinking about nullable's, I wonder if the implicit operator should return `T?` instead?

There's also a null check on the container however I don't think it's needed as if `ContainerLocator.Container` is `null` it will throw an `InvalidOperationException` when being accessed.

dansiegel replied:
No need to return `T?` as the Container would throw an exception rather than return null.

chowarth commented:
Strange, all passing locally 😞 The only thing I can think of is that I removed the creation of the app using the builder in the two test scenarios, so I've added that back.
